### PR TITLE
GH-124284: Add stats for refcount operations on immortal objects

### PR DESCRIPTION
**Pull request by markshannon**
* Issue: gh-124284

### CODE REVIEW

File: Include/internal/pycore_object.h [hunk 1]
```diff
@@ -511,6 +515,7 @@ _Py_TryIncrefFast(PyObject *op) {
     local += 1;
     if (local == 0) {
         // immortal
+        _Py_INCREF_IMMORTAL_STAT_INC()
```
Comment (Fidget-Spinner) [inline Include/internal/pycore_object.h]: Forgot a semicolon here

**Review by Fidget-Spinner — APPROVED** [review]
LGTM. If there's any subtle bugs we can fix them later. It's stats anyways not something mission critical.

**Comment by markshannon** [comment]
Heres' what the stats look like: https://github.com/faster-cpython/benchmarking-public/blob/main/results/bm-20240920-3.14.0a0-e179c31/bm-20240920-azure-x86_64-faster%252dcpython-immortal_ref_count_s-3.14.0a0-e179c31-pystats.md#object-stats

**Review by MandeepSinghPB08 — COMMENTED** [review]
Thanks for helping

**Pull request merged**